### PR TITLE
Gizmo small example

### PR DESCRIPTION
**Pull request by kpal81xd**
- Renames gizmo example to editor (example getting v large)
- Created simple gizmo example
- Added logic for consistent gizmo size based on canvas css dimensions

image

### CODE REVIEW

File: examples/src/examples/misc/gizmo.example.mjs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    app.resizeCanvas();
+    const bounds = canvas.getBoundingClientRect();
+    const dim = camera.camera.horizontalFov ? bounds.width : bounds.height;
+    gizmo.size = 1024 / dim;
+};
+window.addEventListener('resize', resize);
+resize();
+
+// grid lines
+const createGridLines = (size = 1) => {
```
Comment (mvaligursky) [inline examples/src/examples/misc/gizmo.example.mjs:125]: prepare lines in format `app.drawLines` needs, instead of using `drawLine` in a loop.

**Review by mvaligursky — APPROVED** [review]

File: examples/src/examples/misc/gizmo.example.mjs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    clearColor: new pc.Color(0.1, 0.1, 0.1),
+    farClip: 1000
+});
+camera.addComponent('script');
+const orbitCamera = camera.script.create('orbitCamera');
+camera.script.create('orbitCameraInputMouse');
+camera.script.create('orbitCameraInputTouch');
+camera.setPosition(1, 1, 1);
+app.root.addChild(camera);
+orbitCamera.distance = 5 * camera.camera?.aspectRatio;
```
Comment (willeastcott) [inline examples/src/examples/misc/gizmo.example.mjs:89]: Isn't `camera.camera` guaranteed to be valid at this point?

File: examples/src/examples/misc/gizmo.example.mjs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+// create light entity
+const light = new pc.Entity('light');
+light.addComponent('light', {
+    intensity: 1
```
Comment (willeastcott) [inline examples/src/examples/misc/gizmo.example.mjs:94]: This is the default intensity so no need to specify?

**Review by willeastcott — APPROVED** [review]
Approving with minor feedback.

File: examples/src/examples/misc/gizmo.example.mjs (hunk 1 above)
Comment (kpal81xd) [reply examples/src/examples/misc/gizmo.example.mjs:125]: Oops yea forgot about that

File: examples/src/examples/misc/gizmo.example.mjs (hunk 2 above)
Comment (kpal81xd) [reply examples/src/examples/misc/gizmo.example.mjs:89]: It is but typescript doesnt know that

File: examples/src/examples/misc/gizmo.example.mjs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+// scene settings
+app.scene.ambientLight = new pc.Color(0.2, 0.2, 0.2);
+
+// create entities
+const box = new pc.Entity('box');
+box.addComponent('render', {
+    type: 'box',
+    material: createColorMaterial(new pc.Color(0.8, 0.8, 0.8))
```
[the remark this answers is not in the document]
Comment (kpal81xd) [reply examples/src/examples/misc/gizmo.example.mjs:73]: I could do I was considering coloring the box that was all

File: examples/src/examples/misc/gizmo.example.mjs (hunk 3 above)
Comment (kpal81xd) [reply examples/src/examples/misc/gizmo.example.mjs:94]: Just to be explicit but can remove

**Pull request merged**